### PR TITLE
interop fixes

### PR DESCRIPTION
**Pull request by arnetheduck**
* add infinity point initializer for keys
* use bytes8 for domain to avoid endianess issues

### CODE REVIEW

File: blscurve/common.nim [hunk 1]
```diff
@@ -657,33 +658,29 @@ proc mulCoFactor*(point: ECP2_BLS381): ECP2_BLS381 =
   mul(lowpart, G2_CoFactorLow)
   add(result, lowpart)
 
-proc hashToG2*(msgctx: sha256, domain: uint64): ECP2_BLS381 =
+proc hashToG2*(msgctx: sha256, domain: array[8, byte]): ECP2_BLS381 =
```
Comment (mratsim) [inline blscurve/common.nim]: Domain not array[8, byte]

File: blscurve/bls.nim [hunk 2]
```diff
@@ -32,6 +33,12 @@ const
   RawVerificationKeySize* = MODBYTES_384
   RawSignatureSize* = MODBYTES_384 * 2
 
+proc init*(T: SigKey | VerKey): auto =
+  ## Initialize ``SignatureKey``, ``VerificaitonKey`` to the infinitiy point
```
Comment (mratsim) [inline blscurve/bls.nim]: typo: VerificationKey

File: blscurve/bls.nim [hunk 3]
```diff
@@ -32,6 +33,12 @@ const
...
+  var ret: T
+  ret.point.inf()
+  ret
```
Comment (mratsim) [inline blscurve/bls.nim]: When we do `echo SigKey()` we do get the infinity point though. Is there a disconnect between `getBytes`/hex serialization and the actual memory representation?

Comment (arnetheduck) [reply blscurve/bls.nim]: `inf` does stuff: https://github.com/status-im/nim-blscurve/blob/master/blscurve/csources/64/ecp_BLS381.c#L149

I'm not sure how the serialization works exactly, but it seems prudent to make this call here (rust version does a similar thing)

Comment (mratsim) [reply blscurve/bls.nim]: I think when we do SigKey() we get a full zero uncompressed signature, but then when we compress it, we add the compressed bit, which gets serialized as 0xc0000... into hex

**Comment by cheatfate** [comment]
@arnetheduck are we need `stew` package just to keep `deprecated` code working? I think this is overhead.

**Comment by arnetheduck** [comment]
@cheatfate there will be more uses soon enough I suspect, there are several ways the code could be simplified with a few tricks from stew - I prefer that it's used liberally so we don't introduce tech debt unnecessarily

**Pull request merged**